src/storage/src/hummock/iterator/concat_inner.rs

```diff
@@ -16,16 +16,52 @@ use std::cmp::Ordering::{Equal, Greater, Less};
 use std::future::Future;
 use std::sync::Arc;
 
+use itertools::Itertools;
+use risingwave_common::must_match;
 use risingwave_hummock_sdk::key::FullKey;
 use risingwave_hummock_sdk::KeyComparator;
 use risingwave_pb::hummock::SstableInfo;
 
 use crate::hummock::iterator::{DirectionEnum, HummockIterator, HummockIteratorDirection};
 use crate::hummock::sstable::SstableIteratorReadOptions;
+use crate::hummock::sstable_store::TableHolder;
 use crate::hummock::value::HummockValue;
-use crate::hummock::{HummockResult, SstableIteratorType, SstableStoreRef};
+use crate::hummock::{HummockResult, SstableIteratorType, SstableStore, SstableStoreRef};
 use crate::monitor::StoreLocalStatistic;
 
+enum ConcatItem {
+    Unfetched(SstableInfo),
+    Prefetched(TableHolder),
```

Comment:
Since we would preload sstable for bloom-filter in all case, could we only keep `Prefetched` ?

Reply:
> Since we would preload sstable for bloom-filter in all case, could we only keep `Prefetched` ?

If the only call is a `seek` and we choose to seek in the first SST, then we do not need to preload other SSTs.

Reply:
> Since we would preload sstable for bloom-filter in all case, could we only keep `Prefetched` ?

If the only call on this iterator is a seek and we find out that we should seek in the last SST, then we do not need to preload other SSTs.

Reply:
Do you mean that `ReadOptions` does not require `check-bloom-filter`?

Reply:
> Do you mean that `ReadOptions` does not require `check-bloom-filter`?

I mean that in the future another use may copy lots of instances of `ConcatIterators` just for point get

Reply:
How about use
```rust

enum ConcatData {
    Unfetched(Vec<SstableInfo>),
    Prefetched(Vec<TableHolder>),
```

Reply:
> How about use
> 
> ```rust
> enum ConcatData {
>     Unfetched(Vec<SstableInfo>),
>     Prefetched(Vec<TableHolder>),
> ```

same
> If the only call on this iterator is a seek and we find out that we should seek in the last SST, then we do not need to preload other SSTs.